packages/engine/src/simulation/package/state/packages/behavior_execution/chain.rs

```diff
@@ -27,6 +24,50 @@ pub fn gather_behavior_chains<B: Deref<Target = AgentBatch>>(
     })))
 }
 
+pub fn pool_behavior_list_bytes_iter<'a, B: Deref<Target = AgentBatch>>(
+    agent_pool: &'a [B],
+) -> Result<impl Iterator<Item = Vec<&[u8]>> + 'a> {
+    let mut iterables = Vec::with_capacity(agent_pool.len());
+
+    for agent_batch in agent_pool {
+        let iterable = behavior_list_bytes_iter(agent_batch.record_batch()?)?;
+        iterables.push(iterable);
+    }
+    Ok(iterables.into_iter().flatten())
+}
+
+pub fn behavior_list_bytes_iter(
+    agent_batch: &RecordBatch,
+) -> Result<impl Iterator<Item = Vec<&[u8]>>> {
+    let column_name = "behaviors";
+    let row_count = agent_batch.num_rows();
+    let (column_id, _) = agent_batch
+        .schema()
+        .column_with_name(column_name)
+        .ok_or_else(|| crate::datastore::Error::ColumnNotFound(column_name.into()))?;
+    let column = agent_batch.column(column_id);
+    let col_data = column.data_ref();
+
+    let list_indices = unsafe { col_data.buffers()[0].typed_data::<i32>() };
+    let string_indices = unsafe { col_data.child_data()[0].buffers()[0].typed_data::<i32>() };
```

Comment:
Can you add a SAFETY section why this is safe?

Reply:
This came from the merge, but I'll look into it

Reply:
The current main branch should have a SAFETY section. It's basically verifying, that this is truly an `i32`.